core/templates/dev/head/pages/FooterDirective.ts

```diff
@@ -12,14 +12,21 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-/**
- * @fileoverview Data and controllers for the Oppia library footer.
+ /**
+ * @fileoverview Directive for the footer.
  */
 
-oppia.controller('LibraryFooter', [
-  '$scope', '$window', 'LIBRARY_PAGE_MODES', 'LIBRARY_PATHS_TO_MODES',
-  function($scope, $window, LIBRARY_PAGE_MODES, LIBRARY_PATHS_TO_MODES) {
-    var pageMode = LIBRARY_PATHS_TO_MODES[$window.location.pathname];
-    $scope.footerIsDisplayed = (pageMode !== LIBRARY_PAGE_MODES.SEARCH);
-  }
-]);
+require('domain/utilities/UrlInterpolationService.ts');
+
+oppia.directive('oppiaFooter', ['UrlInterpolationService', function(
+    UrlInterpolationService) {
+  return {
+    restrict: 'E',
+    scope: {},
```

Comment:
Instead of creating two directives having duplicate HTML code maybe we can create one directive with attribute `display-hidde=true`. (Not sure whether this will be in the structure you want to accomplish.)

Reply:
Then this would be a change to a functionality I am not targeting right now. I am targeting just the standalone controllers for now (to get the folders-by-feature PR completed) and then we can do this in a separate PR because this can cause regressions and needs testing.